common/utilities/integration_adapters_logger.py

```diff
@@ -0,0 +1,109 @@
+    a high level than INFO but lower than  WARNING
+    """
+    logging.addLevelName(AUDIT, "AUDIT")
+
+    def audit(self, message, *args, **kws):
+        if self.isEnabledFor(AUDIT):
+            self._log(AUDIT, message, args, **kws)
+
+    logging.Logger.audit = audit
+
+
+def load_global_log_config():
+    """
+    A general method to load the overall config of the system, specifically it modifies the root handler to output
+    to sysout and sets the default log levels and format. This is expected to be called once at the start of a
+    system.
+    """
+    _add_audit_level_log()
+    logger = logging.getLogger()
+    logger.setLevel(logging.NOTSET)
+    handler = logging.StreamHandler(sys.stdout)
+    logging.Formatter.converter = time.gmtime
+    formatter = logging.Formatter('[%(asctime)s.%(msecs)03d] '
+                                  '%(message)s pid=%(process)d LogLevel=%(levelname)s ProcessKey=%(processKey)s',
+                                  '%Y-%m-%d %H:%M:%S')
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+
+def _format_values_in_map(dict_values: dict) -> dict:
+    """
+    Replaces the values in the map with key=value so that the key in a string can be replaced with the correct
+    log format, also surrounds the value with quotes if it contains spaces
+    """
+    new_map = {}
+    for key, value in dict_values.items():
+        if ' ' in value:
+            value = f"\"{value}\""
+
+        new_map[key] = f"{key.replace(' ', '')}={value}"
+    return new_map
+
+
+def _formatted_string(message: str, dict_values: dict) -> str:
+    """
+    Populates the string with the correctly formatted dictionary values
+    """
+    formatted_values = _format_values_in_map(dict_values)
+    return message.format(**formatted_values)
+
+
+class IntegrationAdaptersLogger:
+
+    def __init__(self, log_ref: str = "SYS"):
+        self.process_key_tag = log_ref
+        self.logger = logging.getLogger()
+
+    def _write(self, level, message, process_key_num: str):
+        if not process_key_num:
+            process_key_num = "000"
+        self.logger.log(level, message, extra={'processKey': self.process_key_tag + process_key_num})
+
+    def _format_and_write(self, message, values, process_key_num, request_id, correlation_id, level):
+        """
+        Formats the string and appends the appropriate values if they are included before writing the log
+        """
+        message = _formatted_string(message, values)
+
+        if request_id:
+            message += f' RequestId={request_id}'
+        if correlation_id:
+            message += f' CorrelationId={correlation_id}'
+
+        self._write(level, message, process_key_num)
+
+    def info(self, message: str, values: dict = None, process_key_num: str = None,
+             request_id: str = None, correlation_id=None):
+        values = values if values is not None else {}
```

Comment:
You could extract the implementation of all of these log methods to a single method that just takes the log level.

Reply:
As discussed offline, this won't really make things more readable, so I'm happy to leave this as it is.